From 348c202ea55f9ec21ba10ffc16dbcd1c3f8d7cfa Mon Sep 17 00:00:00 2001
From: thardin <thardin@9553f0bf-9b14-0410-a0b8-cfaf0461ba5b>
Date: Tue, 14 Sep 2010 14:45:43 +0000
Subject: [PATCH] Add av_popcount() to libavutil/common.h and bump minor version

[anssi@xbmc.org: backported as required for DTS profile exporting patch]

git-svn-id: file:///var/local/repositories/ffmpeg/trunk@25120 9553f0bf-9b14-0410-a0b8-cfaf0461ba5b
---
 libavutil/common.h |   17 +++++++++++++++++
 2 files changed, 18 insertions(+), 1 deletions(-)

diff --git a/libavutil/common.h b/libavutil/common.h
index d054f87..6404076 100644
--- a/libavutil/common.h
+++ b/libavutil/common.h
@@ -192,6 +192,20 @@ static inline av_const int av_ceil_log2_c(int x)
     return av_log2((x - 1) << 1);
 }
 
+/**
+ * Count number of bits set to one in x
+ * @param x value to count bits of
+ * @return the number of bits set to one in x
+ */
+static inline av_const int av_popcount_c(uint32_t x)
+{
+    x -= (x >> 1) & 0x55555555;
+    x = (x & 0x33333333) + ((x >> 2) & 0x33333333);
+    x = (x + (x >> 4)) & 0x0F0F0F0F;
+    x += x >> 8;
+    return (x + (x >> 16)) & 0x3F;
+}
+
 #define MKTAG(a,b,c,d) ((a) | ((b) << 8) | ((c) << 16) | ((d) << 24))
 #define MKBETAG(a,b,c,d) ((d) | ((c) << 8) | ((b) << 16) | ((a) << 24))
 
@@ -351,3 +365,6 @@ static inline av_const int av_ceil_log2_c(int x)
 #ifndef av_clipf
 #   define av_clipf         av_clipf_c
 #endif
+#ifndef av_popcount
+#   define av_popcount      av_popcount_c
+#endif
-- 
1.7.3

